crates/ruff_python_semantic/src/model.rs

```diff
@@ -878,19 +900,29 @@ impl<'a> SemanticModel<'a> {
     }
 
     /// Returns the current top most scope.
-    pub fn scope(&self) -> &Scope<'a> {
+    pub fn current_scope(&self) -> &Scope<'a> {
         &self.scopes[self.scope_id]
     }
 
+    /// Returns a mutable reference to the current top most scope.
+    pub fn current_scope_mut(&mut self) -> &mut Scope<'a> {
+        &mut self.scopes[self.scope_id]
+    }
+
+    /// Returns an iterator over all scopes, starting from the current scope.
+    pub fn current_scopes(&self) -> impl Iterator<Item = &Scope> {
+        self.scopes.ancestors(self.scope_id)
+    }
+
     /// Returns the parent of the given scope, if any.
-    pub fn parent_scope(&self, scope: &Scope) -> Option<&Scope<'a>> {
+    pub fn scope_parent(&self, scope: &Scope) -> Option<&Scope<'a>> {
```

Comment:
I changed this from `parent_scope` to `scope_parent`... it takes a scope and returns its parent. I don't know if this is an improvement, but in a future PR I added `statement_parent` to take a statement and return its parent, and it felt better to have `thing_parent` than multiple `parent_thing` methods.

Reply:
I prefer `parent_scope`. I find it reads more naturally. I also prefer to name methods after what they return, the arguments only come to play when I want to achieve overloading and need to distinguish the functions.

Reply:
I think `parent_scope()` makes more sense relative to the current scope. `scope_parent` makes more sense when it consumes a scope and gives you _its_ parent 🤷‍♀️ we already went back and forth on this in the pull request that added this though.

Reply:
I legitimately forgot that we'd debated this in the prior PR, sorry to bring it up again. I will leave it as `parent_scope` for now, since we seem undecided and I'll err on the side of inertia.